### PR TITLE
Rips out the ability to execute a Groovy script with this plugin

### PR DESCRIPTION
**Pull request by seanturner83**
(see SECURITY-294)

### CODE REVIEW

**Comment by seanturner83** [comment]
This could be a breaking change for some, but I can't see how a plugin that allows arbitrary code execution can be meaningfully locked down in an ecosystem this large.

**Comment by daniel-beck** [comment]
@seanturner83 Ideally you'd add a compatibility warning for this plugin, see https://wiki.jenkins-ci.org/display/JENKINS/Marking+a+new+plugin+version+as+incompatible+with+older+versions

This should make users at least read the changelog, in which this change should be very prominent.

Also, related fields (the script String) could be made transient; that way, there's no compatibility warning when the field is removed, but it won't be written back to disk.

**Comment by seanturner83** [comment]
@jenkinsci/code-reviewers

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
         <artifactId>plugin</artifactId>
-        <version>1.565.1</version>
+        <version>1.585</version>
     </parent>
 
     <artifactId>artifactdeployer</artifactId>
     <packaging>hpi</packaging>
     <name>Jenkins Artifact Deployer Plug-in</name>
-    <version>0.34-SNAPSHOT</version>
+    <version>0.40</version>
```
Comment (oleg-nenashev) [inline pom.xml]: Probably it should be 1.0

File: pom.xml [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
     <url>http://wiki.jenkins-ci.org/display/JENKINS/ArtifactDeployer+Plugin</url>
```
Comment (oleg-nenashev) [inline pom.xml]: Should also declare incompatibility: https://wiki.jenkins.io/display/JENKINS/Marking+a+new+plugin+version+as+incompatible+with+older+versions

**Pull request merged**